### PR TITLE
Fixed empty snapshot desc issue. Added err display for fileshare api calls, removed commented code from profile

### LINKED ISSUE #440: Modify fileshare - empty description returns error in API, user is not shown any notification
**Issue/Feature Description:**
When the user modifies the on premise file share and gives empty description, the API returns error that description cannot be empty. The user is not shown any notification and has no way of knowing what the error is.

image

**Why this issue to fixed / feature is needed(give scenarios or use cases):**

**How to reproduce, in case of a bug:**
1. Go to On premise file storage.
2. Create a fileshare.
3. Click on modify in the operation menu.
4. enter name only. No description. click ok
5. Screen remains static. error in the API response
**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind bug fix

**What this PR does / why we need it**:
This PR fixes the issue of empty description throwing error while modifying the file share.
This PR also adds success and error notifications for the Fileshare CRUD, Snapshot CRUD and ACL CRUD.

**Which issue(s) this PR fixes**:
Fixes #440 

**Test Report Added?**:
/kind TESTED
> /kind NOT-TESTED

**Test Report**:
image

image

ACL Success
image

ACL Delete Success
image

Snapshot update success
image

Snapshot delete success
image

File share delete success
image

Fileshare delete error
image

**Special notes for your reviewer**:

### CODE REVIEW

File: src/app/business/block/file-share-detail/file-share-detail.component.ts [hunk 1]
```diff
@@ -352,11 +352,22 @@ export class FileShareDetailComponent implements OnInit{
         this.SnapshotService.deleteSnapshot(id).subscribe((res)=>{
             this.getSnapshots(this.fromFileShareId);
             this.showCreateSnapshot = true;
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: "Fileshare snapshot will be deleted shortly."});
```
Comment (skdwriting) [inline src/app/business/block/file-share-detail/file-share-detail.component.ts:356]: Do we update the real success information later...? How the user will know whether it is really deleted or not.

File: src/app/business/block/file-share-detail/file-share-detail.component.ts [hunk 2]
```diff
@@ -352,11 +352,22 @@ export class FileShareDetailComponent implements OnInit{
...
+            this.msgs = [];
+            this.showCreateSnapshot = true;
+            this.msgs.push({severity: 'error', summary: 'Error', detail: err.message ? err.message : err.json().message});
         })
     }
     deleteAcls(aclId){
         this.FileShareAclService.deleteFileShareAcl(aclId).subscribe((res)=>{
             this.getAcls();
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: "Fileshare ACL will be deleted shortly."});
```
Comment (skdwriting) [inline src/app/business/block/file-share-detail/file-share-detail.component.ts:367]: Fileshare ACL ? I think the message is not proper?

File: src/app/business/block/file-share-detail/file-share-detail.component.ts [hunk 3]
```diff
@@ -433,6 +454,12 @@ export class FileShareDetailComponent implements OnInit{
         this.FileShareAclService.updateFileShareAcl(this.modifyAcl.id,param).subscribe((res)=>{
             this.getAcls();
             this.aclModifyShow = false;
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: "Fileshare ACL updated successfully."});
```
Comment (skdwriting) [inline src/app/business/block/file-share-detail/file-share-detail.component.ts:458]: Fileshare ACL?

File: src/app/business/block/file-share-detail/file-share-detail.component.ts (hunk 1 above)
Comment (anvithks) [reply src/app/business/block/file-share-detail/file-share-detail.component.ts:356]: The message is will be deleted shortly as the API is async. When the list is refreshed again the snapshot may not be visible. Sometimes the snapshot is deleted immediately and the list is refreshed before showing the user.

The dashboard works in a pull mode. So there is no way to update it dynamically.

File: src/app/business/block/file-share-detail/file-share-detail.component.ts (hunk 2 above)
Comment (anvithks) [reply src/app/business/block/file-share-detail/file-share-detail.component.ts:367]: This is for deleting File share access. Do we call it "fileshare access"?
@kumarashit @NajmudheenCT

**Comment by kumarashit** [comment]
Actually its not access level but access permissions. Also, the IPAddr is should not be the access/ACL name. Either something wrong returned from backend or we could show proper message

**Comment by anvithks** [comment]
> Actually its not access level but access permissions. Also, the IPAddr is should not be the access/ACL name. Either something wrong returned from backend or we could show proper message

It is possible to allow only one access permission per IP address. The Access is of type `ip` although there is no indication of what the other types can be.
The API is called Fileshare ACLs.
I think we should take this as a backlog since we will have to relook at the API naming convention and the supported parameters. The API spec does not have any additional information.

GET
image

POST
image

**Review by skdwriting — APPROVED** [review]
LGTM

**Comment by kumarashit** [comment]
Opened an issue.
Ideally the Fileshare ACL is provided for an IP or all IP. The message will be confusing if we show (*). Also, it's not level, but either the Permission or Capability.
It's better to correct it

**Review by kumarashit — APPROVED** [review]
LGTM
(There is an issue opened for fixing this properly and more user friendly)

**Pull request merged**